diff --git a/node_modules/@oclif/core/lib/interfaces/parser.d.ts b/node_modules/@oclif/core/lib/interfaces/parser.d.ts
index acb8583..2d53745 100644
--- a/node_modules/@oclif/core/lib/interfaces/parser.d.ts
+++ b/node_modules/@oclif/core/lib/interfaces/parser.d.ts
@@ -200,6 +200,7 @@ export type OptionFlagProps = FlagProps & {
     helpValue?: string;
     options?: readonly string[];
     multiple?: boolean;
+    singleValue?: boolean;
     /**
      * Delimiter to separate the values for a multiple value flag.
      * Only respected if multiple is set to true. Default behavior is to
diff --git a/node_modules/@oclif/core/lib/parser/parse.js b/node_modules/@oclif/core/lib/parser/parse.js
index 032063d..72d2ed5 100644
--- a/node_modules/@oclif/core/lib/parser/parse.js
+++ b/node_modules/@oclif/core/lib/parser/parse.js
@@ -145,7 +145,7 @@ class Parser {
                     continue;
                 }
             }
-            if (parsingFlags && this.currentFlag && this.currentFlag.multiple) {
+            if (parsingFlags && this.currentFlag && this.currentFlag.multiple && !this.currentFlag.singleValue) {
                 this.raw.push({ flag: this.currentFlag.name, input, type: 'flag' });
                 continue;
             }
