diff --git a/node_modules/pkg/prelude/bootstrap.js b/node_modules/pkg/prelude/bootstrap.js
index 034bd15..5913878 100644
--- a/node_modules/pkg/prelude/bootstrap.js
+++ b/node_modules/pkg/prelude/bootstrap.js
@@ -700,7 +700,14 @@ function payloadFileSync(pointer) {
         fs.rmdirSync(folder, { recursive: true });
       }
     } else {
-      fs.rmSync(folder, { recursive: true });
+      try {
+        fs.rmSync(folder, { recursive: true,  });
+      } catch (e){
+        if (e.code !== "ENOENT"){
+          throw e
+        }
+      }
+      
     }
   }
   const temporaryFiles = {};
